x-pack/plugins/ml/public/embeddables/anomaly_explorer/anomaly_explorer_embeddable_factory.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+import type { StartServicesAccessor } from 'kibana/public';
+
+import type {
+  EmbeddableFactoryDefinition,
+  IContainer,
+} from '../../../../../../src/plugins/embeddable/public';
+import { HttpService } from '../../application/services/http_service';
+import type { MlPluginStart, MlStartDependencies } from '../../plugin';
+import type { MlDependencies } from '../../application/app';
+import {
+  ANOMALY_EXPLORER_CHARTS_EMBEDDABLE_TYPE,
+  AnomalyExplorerEmbeddableInput,
+  AnomalyExplorerEmbeddableServices,
+} from '..';
+import { AnomalyExplorerChartsService } from '../../application/services/anomaly_explorer_charts_service';
+
+export class AnomalyExplorerEmbeddableFactory
```

Comment:
Initially this is only creating an embeddable for anomaly explorer charts. Should it (and the file) be renamed to match the `type` of the embeddable, such as `AnomalyExplorerChartsEmbeddableFactory` or just `AnomalyChartsEmbeddableFactory`? If we eventually wrap multiple Anomaly Explorer components inside a container (such as swim lane, charts and table), that could then become the `AnomalyExplorerEmbeddableFactory`

Reply:
Renamed here https://github.com/elastic/kibana/pull/94396/commits/3c31e37dacc795ad9e4b0b71d459616bda58b979